Ingress/controllers/nginx-third-party/nginx.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/golang/glog"
+)
+
+type nginxConfiguration struct {
```

Comment:
I'd suggest more documentation around what is allowed not allowed in these fields. Or if they're exactly the same as nginx fields, please add a comment with a link to the docs above the struct.